### PR TITLE
Disallow inferred splices and set the correct position

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Addressing comments in #4775

File: tests/neg/i4774a.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+import scala.quoted._
+
+object Test {
+  def loop[T](x: Expr[T])(implicit t: Type[T]): Expr[T] = '{
+    val y: ~t = ~x
+    ~loop( // error: inferred loop[~t] where T should be used
```
Comment (smarter) [inline tests/neg/i4774a.scala]: Instead of giving an error, couldn't the typechecker be smart enough to dealias?

Comment (nicolasstucki) [reply tests/neg/i4774a.scala]: The difficulty is that this depends on the quotation level and we do not have it at typer time.

Comment (nicolasstucki) [reply tests/neg/i4774a.scala]: I wouldn't worry to much about this case and encourage the use of `T` instead of `~t`.

**Comment by biboudis** [comment]
Looks good!

**Pull request merged**